### PR TITLE
Do/meeting page

### PR DESCRIPTION
**Pull request by Dahvd**
## Meeting Page

I created three components for the meeting page.

## MeetingNameDate
Has the header for the meeting
## MeetingTopics 
Has the field for all the topics 
## Attendies
Has field for participants

I know this is just visual so I probably have to do a lot with props and stuff but ill be at work for 5 hours so any feedback is appreciated

# Screenshot 
image

### CODE REVIEW

**Comment by zbarnz** [comment]
Screenshots?

**Comment by zbarnz** [comment]
Screenshots or riot

File: package.json [hunk 1]
```diff
@@ -16,5 +16,6 @@
   "bugs": {
     "url": "https://github.com/SocexSolutions/agenda-v2/issues"
   },
-  "homepage": "https://github.com/SocexSolutions/agenda-v2#readme"
+  "homepage": "https://github.com/SocexSolutions/agenda-v2#readme",
+  "dependencies": {}
```
Comment (thudsonbu) [inline package.json:20]: Can you undo these changes?

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "agenda",
+  "version": "0.2.0",
+  "lockfileVersion": 1
+}
```
Comment (thudsonbu) [inline package-lock.json:5]: Can you also get rid of the package-lock.json

File: www/components/Attendies.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import styles from "../styles/Attendies.module.css";
+import Chip from "./Chip";
+import Input from "./Input";
+function Attendies() {
```
Comment (thudsonbu) [inline www/components/Attendies.js:4]: Can you add a newline betweed the imports and the component?

File: www/components/Button.js [hunk 4]
```diff
@@ -29,28 +29,28 @@ const Button = ({ icon, text, size, onClick, varient, stretch }) => {
...
-    case "menu":
-      className += " " + styles.menu;
-      break;
-    case "topic":
-      className += " " + styles.topic;
-      break;
-    case "icon":
-      className += " " + styles.icon;
-      break;
+  switch ( varient ) {
```
Comment (thudsonbu) [inline www/components/Button.js:32]: Try not to change files that you aren't really making changes to.

File: www/components/Chip.js [hunk 5]
```diff
@@ -1,13 +1,18 @@
 import PropTypes from "prop-types";
 import styles from "../styles/Chip.module.css";
-import CancelRoundedIcon from "@material-ui/icons/CancelRounded";
+import CancelOutlined from "@material-ui/icons/CancelOutlined";
+import AccountCircleIcon from "@material-ui/icons/AccountCircle";
+const Chip = ({ text, editing, deleteFunc, icon }) => {
```
Comment (thudsonbu) [inline www/components/Chip.js:5]: Can you add a newline between the imports and the function?

File: www/components/Attendies.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  return (
+    <div className={styles.attbox}>
+      <div className={styles.attW}>
+        <h2>Attendies</h2>
+        Owner
+        <Chip text="Tom Hudson" icon={true}/>
+        <br />
+        Participant(s)
+        <div className={styles.atts}>
```
Comment (thudsonbu) [inline www/components/Attendies.js:13]: It is generally a good idea to have all text within its own `<p>` or `<h>` tag so that if there are global styles for those tags they take effect. Can you wrap those titles in something? Also, people generally don't like to see `<br>` tags as they are inconsistend between browsers.

File: www/components/Chip.js [hunk 7]
```diff
@@ -1,13 +1,18 @@
...
+  if ( icon == true ) {
+    icon=<AccountCircleIcon className={styles.icons} />;
+  }
```
Comment (thudsonbu) [inline www/components/Chip.js:8]: Thanks for fixing sir!

File: www/components/MeetingNameDate.js [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  return (
+    <div className={styles.meetingBar}>
+      <FlagIcon style={{ color: "var(--agendaSecondary)", fontSize: 37 }}/>
+      <form className={styles.form}>
+          <div>
+            <Input placeholder="Meeting Name" />
+          </div>
+          <div className={styles.date}>
+            <Input placeholder="mm/dd/yy hh:mm"  />
+          </div>
```
Comment (thudsonbu) [inline www/components/MeetingNameDate.js:23]: Instead of wrapping the `Input` components in `div`s like this you can select the `Input` in the CSS like this because it is within the `form` class:


```
.form Input {
  mystyles;
}
```

File: www/components/MeetingNameDate.js [hunk 9]
```diff
@@ -0,0 +1,29 @@
+/* eslint-disable indent */
+/* eslint-disable no-trailing-spaces */
+// David
+// This component has a flag icon which is white while meeting
+// is being edited and red when saved,
```
Comment (thudsonbu) [inline www/components/MeetingNameDate.js:5]: The flag icon indicates the priority of the meeting, maybe we should highlight it differently if it is being edited or just add "editing" in front of the title.

File: www/components/Participant.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+import PropTypes from "prop-types";
+import Chip from "./Chip";
+import styles from "../styles/Participant.module.css";
+
+function Participant({ text, owner }) {
+  return (
+    <div>
+      <Chip className={styles.text} text={text} editing={true}
+        deleteFunc={true} icon={true}/>
```
Comment (thudsonbu) [inline www/components/Participant.js:10]: When there are lots of props do something like this:


```
<Chip
  className={styles.text}
  text={text}
  editing={true}
  deleteFunc={true}
  icon={true}
/>
```

File: www/components/MeetingNameDate.js (hunk 8 above)
Comment (thudsonbu) [inline www/components/MeetingNameDate.js:23]:
```suggestion
        <div>
          <Input placeholder="Meeting Name" />
        </div>
        <div className={styles.date}>
          <Input placeholder="mm/dd/yy hh:mm"  />
        </div>
```
Just improving the indent a bit.

File: www/components/MeetingTopics.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+import styles from "../styles/MeetingTopics.module.css";
+import Chip from "./Chip";
+import Input from "./Input";
+function MeetingTopics() {
```
Comment (thudsonbu) [inline www/components/MeetingTopics.js:4]:
```suggestion
import Input from "./Input";

function MeetingTopics() {
```

File: www/components/MeetingTopics.js [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <div className={styles.topicbox}>
+      <div className={styles.topicsW}>
```
Comment (thudsonbu) [inline www/components/MeetingTopics.js:7]: The classname `topicsW` doesn't really tell you a whole lot about what the class does. Also `css` classes are usually done using kabab case aka `topics-w`.

File: www/pages/meeting.js [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+import MeetingNameDate from "../components/MeetingNameDate";
+import MeetingTopics from "../components/MeetingTopics";
+import Attendies from "../components/Attendies";
+
+function meeting() {
+  return (
+    <div>
+      <MeetingNameDate />
+      <MeetingTopics />
+      <Attendies />
```
Comment (thudsonbu) [inline www/pages/meeting.js:11]: It doesn't get much easier to read then this 👍

File: www/styles/Chip.module.css [hunk 14]
```diff
@@ -14,10 +14,11 @@
   font-size: 18px;
   padding: 0px;
   margin: 7px;
+  color: var(--textBlue)
 }
 
 .deleteIcon {
-  color: var(--agendaSecondary);
+  color: var(--danger);
```
Comment (thudsonbu) [inline www/styles/Chip.module.css:21]: Idk I thought the red might be to aggressive.

File: www/styles/Chip.module.css [hunk 15]
```diff
@@ -26,3 +27,10 @@
   cursor: pointer;
   opacity: .8;
 }
+
+
+.icons {
+  color: var(--skyBlue);
+  display: inline-block;
+  margin-left: .5rem;
```
Comment (thudsonbu) [inline www/styles/Chip.module.css:35]: I don't like using `rem` as a unit because if we change the global font size it changes this. Its kind of controversial though.

File: www/styles/MeetingNameDate.module.css [hunk 16]
```diff
@@ -0,0 +1,16 @@
+.meetingBar {
+    display: flex;
+    margin-bottom: 50px;
+    margin-left: 50px;
+    margin-right: 50px;
+
```
Comment (thudsonbu) [inline www/styles/MeetingNameDate.module.css:6]:
```suggestion
```

File: www/styles/MeetingNameDate.module.css [hunk 17]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+.form{ width: 100%;
+    margin-left: 10px;
+    display: flex;
+}
+
+.date {
+margin-left: auto;
```
Comment (thudsonbu) [inline www/styles/MeetingNameDate.module.css:15]:
```suggestion
  margin-left: auto;
```

File: www/styles/MeetingNameDate.module.css [hunk 18]
```diff
@@ -0,0 +1,16 @@
...
+    margin-bottom: 50px;
+    margin-left: 50px;
+    margin-right: 50px;
+
+}
+
+.form{ width: 100%;
+    margin-left: 10px;
+    display: flex;
+}
```
Comment (thudsonbu) [inline www/styles/MeetingNameDate.module.css:12]:
```suggestion
.form{ 
    width: 100%;
    margin-left: 10px;
    display: flex;
}
```

Can you change your default tab size to two spaces?

File: www/styles/MeetingTopics.module.css [hunk 19]
```diff
@@ -0,0 +1,19 @@
...
+    background-color: #0E2D68;
+    color: white;
+    margin-left: 50px;
+    margin-right: 50px;
+    margin-bottom: 20px;
+}
+
+.topicsW{
+    margin-left: 20px;
+
```
Comment (thudsonbu) [inline www/styles/MeetingTopics.module.css:11]:
```suggestion
```

**Review by thudsonbu — CHANGES_REQUESTED** [review]
Looks pretty good overall 👍. I think that this should be split up into multiple PRs though for each component.

File: package.json (hunk 1 above)
Comment (Dahvd) [reply package.json:20]: yes idek what i did to do that, just those two lines?

File: www/components/MeetingNameDate.js (hunk 9 above)
Comment (Dahvd) [reply www/components/MeetingNameDate.js:5]: ok i just mistook why it was white in one figma and red in another.  i dont think we need to say editing it will be obvious

File: www/components/MeetingNameDate.js (hunk 8 above)
Comment (Dahvd) [reply www/components/MeetingNameDate.js:23]: I have two inputs and i want each one styled differently because of the spacing

File: www/components/Participant.js (hunk 10 above)
Comment (Dahvd) [reply www/components/Participant.js:10]: Just saying, prettier did that for me

Comment (Dahvd) [reply www/components/Participant.js:10]: But you just had to replace with BS lint

File: www/styles/Chip.module.css (hunk 15 above)
Comment (Dahvd) [reply www/styles/Chip.module.css:35]: are you not the one who did this lmao

**Review by thudsonbu — APPROVED** [review]

**Review by zbarnz — APPROVED** [review]
